ts/client/src/accounts/bank.ts

```diff
@@ -481,6 +475,39 @@ export class Bank implements BankForHealth {
   getDepositRateUi(): number {
     return this.getDepositRate().toNumber() * 100;
   }
+
+  getNetBorrowLimitPerWindow(): I80F48 {
+    return I80F48.fromI64(this.netBorrowLimitPerWindowQuote).div(this.price);
+  }
+
+  getNetBorrowLimitPerWindowUi(): number {
+    return toUiDecimals(
+      I80F48.fromI64(this.netBorrowLimitPerWindowQuote).div(this.price),
+      this.mintDecimals,
+    );
+  }
+
+  getMaxWithdrawNative(vaultBalance: BN, userDeposits = ZERO_I80F48()): I80F48 {
+    // borrow is capped to minVaultToDepositsRatio and netBorrowLimitPerWindowQuote
+    const maxWithdrawFromVault = this.nativeDeposits().mul(
+      I80F48.fromNumber(1 - this.minVaultToDepositsRatio),
+    );
+
+    // userDeposits can exceed maxWithdrawFromVault
+    let maxBorrow = maxWithdrawFromVault.sub(userDeposits).min(ZERO_I80F48());
+    maxBorrow = maxBorrow.min(this.getNetBorrowLimitPerWindow());
```

Comment:
> This has no effect, because it follows a .min(0) in the previous line and net borrow limit is >= 0.
good catch, i meant to do a  .max. changed now


> Also, shouldn't this work with the net borrow limit that's currently remaining, not the maximum in the window?
good catch